### PR TITLE
Shutdown Screen to compose

### PR DESCRIPTION
**Pull request by Gotalicp**
Changed Shutdown fragment to compose
Reworked both shutdownmodels to fit compose

Old:

New:

### CODE REVIEW

**Review by trifonov-i — APPROVED** [review]

File: app/src/main/java/bg/zahov/app/ServiceErrorStateViewModel.kt [hunk 1]
```diff
@@ -1,36 +1,50 @@
...
+                    ServiceState.Unavailable -> _serviceState.update {
+                        it.copy(navigateToShuttingDown = true)
+                    }
+
+                    ServiceState.Shutdown -> _serviceState.update { it.copy(shutDown = true) }
                     else -> {}
                 }
             }
         }
     }
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/ServiceErrorStateViewModel.kt:49]: This block can be removed. You can simplify the code by using the same `ServiceState` in the view layer, or by mapping the flow to the domain-layer model `ShutDownData()` when declaring the `serviceState` flow.
e.g. 

```
val serviceState: StateFlow<ShutDownData> = serviceErrorHandler.observeServiceState()
.map {  } // your transformation
.launchIn(viewModelScope)
```

File: app/src/main/java/bg/zahov/app/ui/error/ShuttingDownScreen.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.app.ui.theme.FitnessTheme
+import bg.zahov.fitness.app.R
+import kotlinx.coroutines.delay
+
+@Composable
+fun ShuttingDownScreen(viewModel: ShuttingDownViewModel = viewModel()) {
+    val uiState = viewModel.state.collectAsStateWithLifecycle()
+    LaunchedEffect(uiState.value) {
+        delay(1000)
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/ui/error/ShuttingDownScreen.kt:33]: Why is this needed? This indicates you have some threading issues.

File: app/src/main/java/bg/zahov/app/ServiceErrorStateViewModel.kt (hunk 1 above)
Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/ServiceErrorStateViewModel.kt:49]: I will just observe serviceError in main activity

File: app/src/main/java/bg/zahov/app/ui/error/ShuttingDownScreen.kt (hunk 2 above)
Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/ui/error/ShuttingDownScreen.kt:33]: i dont know why its in the ui, now its in the viewmodel

**Comment by dimitarstoyanovtumba** [comment]
Out of simple curiosity - why is there a doomsday countdown feature in the app?
